src/sentry/tasks/data_export.py

```diff
@@ -1,8 +1,130 @@
 from __future__ import absolute_import
+
+import csv
+
+from sentry.constants import ExportQueryType
 from sentry.tasks.base import instrumented_task
 
+SNUBA_MAX_RESUILTS = 1000
+
+
+def serialize_issue_by_tag(key, item):
+    result = {
+        "value": item.value,
+        "times_seen": item.times_seen,
+        "last_seen": item.last_seen.strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
+        "first_seen": item.first_seen.strftime("%Y-%m-%dT%H:%M:%S.%fZ"),
+    }
+    if key == "user":
+        euser = item._eventuser
+        result["id"] = euser.ident if euser else ""
+        result["email"] = euser.email if euser else ""
+        result["username"] = euser.username if euser else ""
+        result["ip_address"] = euser.ip_address if euser else ""
+    return result
+
 
 @instrumented_task(name="sentry.tasks.data_export.compile_data", queue="data_export")
 def compile_data(data_export):
-    # TODO(Leander): Implement logic to compile CSV data via provided ExportedData object
+    if data_export.query_type == ExportQueryType.DISCOVER_V2:
+        # TODO(Leander): Implement logic for discover v2 CSVs
+        return
+    elif data_export.query_type == ExportQueryType.BILLING_REPORT:
+        # TODO(Leander): Implement logic for billing report CSVs
+        return
+    elif data_export.query_type == ExportQueryType.ISSUE_BY_TAG:
+        process_issue_by_tag(data_export.query_info)
+        return
+
+
+def process_discover_v2(data_export):
+    return
+
+
+def process_billing_report(data_export):
+    return
+
+
+def process_issue_by_tag(payload):
+    from sentry import tagstore
+    from sentry.models import EventUser, Group, Project, get_group_with_redirect
```

Comment:
@leeandher is importing here necessary?

Reply:
I put the imports specific to whatever process function is running (i.e. When i get to Discover Processing, it'll have different imports specific to Discover.)

Reply:
@leeandher In general, we should put imports at the top of the file. Sometimes this causes errors which is why we do it in a function.

Reply:
alright, will do